tooling/metamodel-generator/src/main/java/org/hibernate/processor/HibernateProcessor.java

```diff
@@ -714,7 +728,7 @@ private void writeIndex() {
 					.createResource(
 							StandardLocation.SOURCE_OUTPUT,
 							ENTITY_INDEX,
-							entityName,
+							URLEncoder.encode(entityName, UTF_8),
```

Comment:
Why is this necessary?

Reply:
Take look at the `Person` in `org.hibernate.orm.test.hql.QuotedIdentifierTest`. There entity name is `The Person` (guess that this is not forbidden by the specification) and processing (and/or compilation) is crashing with message:

`java.lang.IllegalArgumentException: Invalid relative name: The Person`

If you like, I can write similar test for Hibernate Processor. Of course, if such entity name is allowed.

Reply:
OK, I see. But then I think we need to make the same change in `ProcessorSessionFactory.findEntityByUnqualifiedName()`?

>  if such entity name is allowed.

Well that is a _great_ question. It looks like the JPA spec is silent on the format of entity names, except to say that they are case-sensitive, and unique in a PU.

I think, however, that we can infer that they are not supposed to have spaces in them, because then we would not be able to parse the query language.

I think I should open an issue against the spec.

Reply:
No, `ProcessorSessionFactory.findEntityByUnqualifiedName()` will work fine, since entity name is key for `entityNameMappings`. And such entity can be, despite its really 'ugly' name, used in HQL queries if properly quoted. `QuotedIdentifierTest` is testing exactly that. I've just tried to name class 'The Person?', and it is working fine. 
But, I guess that spec should be changes anyway. Although, I guess that some hardcore Star Trek fans will be disappointed if not able to name entites using Klingon alphabet :-) [It is working now, BTW]

Reply:
I've removed that change and excluded all "troublemakers". Unfortunately, I can not find wither where this entity index is specified or used, Problem with such "ugly" entity names can not be solved in Hibernate, because finally it is used in  `com.sun.tools.javac.file.JavacFileManager#getFileForOutput`. That name must be encoded to pass as constructor parameter for `java.net.URI`, but must not be encoded to allow file to be created with proper entity name. Something that Joseph Keller called "Catch 22" :-)

Reply:
I don't think we need to worry about indexing such pathological entity names. They can't be used in queries anyway (since the query can't be parsed).

Reply:
https://github.com/jakartaee/persistence/issues/686

For the record.

Reply:
Can be used if quoted. But I agree that that such names should not be used. Maybe only to put some disclaimer about that for those who like to sue company for mot specifically saying that cat can not be put in microwave oven to dry? :-)

Reply:
It seems that Hibernate will happily accept non-ASCII letters in entity (and/or class) names (PostgreSQL even in column names), so only it should be not big "sacrifice" to not allow special characters

Reply:
> Can be used if quoted. 

Ahhh uuhmmm, yeah, OK sure. I forgot HQL has quoted identifiers (JPQL doesn't).